six/modules/c++/cphd/include/cphd/Data.h

```diff
@@ -20,86 +20,351 @@
+                   numVectors == other.numVectors &&
+                   numSamples == other.numSamples &&
+                   signalArrayByteOffset == other.signalArrayByteOffset &&
+                   pvpArrayByteOffset == other.pvpArrayByteOffset &&
+                   compressedSignalSize == other.compressedSignalSize;
+        }
+        bool operator!=(const Channel& other) const
+        {
+            return !((*this) == other);
+        }
 
-    // 1-based channel number
-    size_t getNumVectors(size_t channel) const;
+        //! Getter functions
+        size_t getNumVectors() const
+        {
+            return numVectors;
+        }
+        size_t getNumSamples() const
+        {
+            return numSamples;
+        }
+        size_t getSignalArrayByteOffset() const
+        {
+            return signalArrayByteOffset;
+        }
+        size_t getCompressedSignalSize() const
+        {
+            return compressedSignalSize;
+        }
 
-    // 1-based channel number
-    size_t getNumSamples(size_t channel) const;
+        //! String that uniquely identifies the CPHD channel
+        std::string identifier;
 
-    size_t getNumBytesPerSample() const;
+        //! Number of vectors in the signal array
+        size_t numVectors;
 
-    size_t getNumBytesVBP() const
+        //! Number of samples per vector in the signal array
+        size_t numSamples;
+
+        //! Signal Array offset from the start of the Signal
+        //! block (in bytes) to the start of the Signal Array
+        //! for the channel.
+        size_t signalArrayByteOffset;
+
+        //! PVP Array offset from the start of the PVP block
+        //! (in bytes) to the start of the PVP Array for the
+        //! channel.
+        size_t pvpArrayByteOffset;
+
+        //! (Optional) Size (in bytes) of the compressed signal array
+        //! byte sequence for the data channel.
+        size_t compressedSignalSize; // Optional
+    };
+
+    /*
+     *  \func Data
+     *
+     *  \brief Default constructor
+     */
+    Data();
+
+    //! Equality operator
+    bool operator==(const Data& other) const
     {
-        return numBytesVBP;
+        return signalArrayFormat == other.signalArrayFormat &&
+               numBytesPVP == other.numBytesPVP &&
+               signalCompressionID == other.signalCompressionID &&
+               channels == other.channels &&
+               supportArrayMap == other.supportArrayMap &&
+               mOffsetMap == other.mOffsetMap;
     }
-
-    void setNumBytesVBP(size_t n)
+    bool operator!=(const Data& other) const
     {
-        numBytesVBP = n;
+        return !((*this) == other);
     }
 
-    bool operator==(const Data& other) const
+    //! Getter functions
+    virtual size_t getNumVectors(size_t channel) const
+    {
+        if (channel < channels.size())
+        {
+            return channels[channel].getNumVectors();
+        }
+        std::ostringstream ostr;
```

Comment:
Might be useful to have some `verifyChannelIsInRange()` method since you're doing this so often